pkg/lib/cockpit-components-terminal.tsx

```diff
@@ -261,19 +357,20 @@
     }
 
     componentWillUnmount() {
-        this.disconnectChannel();
-        this.terminal.dispose();
         window.removeEventListener('resize', this.onWindowResize);
         this.onFocusOut();
+        this.unmountTerminal(this.terminal_state);
+        if (!this.props.state)
+            this.terminal_state.close();
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21680-2a204949-20250814-103842-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)